### DISCUSSION TITLE
Problem with Parsing.Stream typealias

### DISCUSSION THREAD

**Discussion by sillygoose** [pointfreeco/swift-parsing#281] [General]
I am accessing a Bluetooth device and it uses the StreamDelegate protocol which defines Stream, when I added the Parsing import I received errors in my StreamDelegate:
```
import Foundation
import Parsing

class SomeBluettothDevice: NSObject {
    func hasBytesAvailable() -> Void {
    }
    func hasSpaceAvailable() -> Void {
    }
}

extension SomeBluettothDevice: StreamDelegate {
    func stream(_ aStream: Stream, handle eventCode: Stream.Event) {
        switch eventCode {
        case Stream.Event.hasBytesAvailable:
            hasBytesAvailable()
        case Stream.Event.hasSpaceAvailable:
            hasSpaceAvailable()
        default:
            break
        }
    }
}
```
Thought this was weird but I found that Parsing defines its own Stream structure and a typealias for Stream.  Is this a problem or expected behavior that I have to work around?

**Comment by sillygoose** [comment]
I had worked around this by not calling a parser from the module reading the Bluetooth stream but now that I took a further look the easiest fix would have been to add the library name to the code:

```
func stream(_ aStream: Foundation.Stream, handle eventCode: Foundation.Stream.Event) {
...
```

**Reply by tgrapperon** [reply]
Ah yeah, @sillygoose, sorry for not spotting your message sooner. What you did is called "qualifying" the type's name. This can be used when names are ambiguous.
As a trick, you can locally "shadow" ambiguities using a typealias:
```swift
typealias Stream = Foundation.Stream
```
Because this thing is defined in your current module, it takes precedence over the other ones, and allows to directly use `Stream` in your code. You can make it fileprivate if required (and if it doesn't shows into exposed APIs arguments).

**Reply by sillygoose** [reply]
Thanks for the reply, always something new to learn every day (and tough when you don't every day).  My old brain is wired to think the compiler would complain but once I found out about namespaces I see how it all works.
